app/controllers/concerns/accountable.rb

```diff
@@ -14,8 +14,27 @@ def create_user_account
     log e
   end
 
+  def update_account_login_stats
+    return unless login_stats.present? && login_type.in?(AccountLoginStat::LOGIN_TYPES)
+
+    login_stats.update("#{login_type}_at" => Time.zone.now)
+  end
+
   private
 
+  def login_stats
+    @login_stats ||=
+      if @current_user.account.blank?
```

Comment:
It might be interesting to have a log or Sentry error when account is blank off a current_user. We could monitor and see if this ever happens (which it probably shouldn't), and if we can confirm it never happens, maybe we can make the assumption that a user has an account.

You could potentially do that in this PR, or a future PR, if you decided to add that, either way works IMO

Reply:
Oh wait, I see in `AfterLoginJob`, we try to create an account right before adding a record in `AccountLoginStats`, it's a log with `cannot_create_unique_account_record` if an Account isn't created. We could look that up in Sentry and see if we ever hit it

Reply:
I looked this up. There have been 219 examples in the last 30 days, so failing to create an account is something that can definitely happen.